internal/provider/statuspage.go

```diff
@@ -9,7 +9,7 @@ import (
 	"io"
```

Comment:
We can deduplicate this code by writing a method that converts the diagnostics. I would use this file as the default implementation, and have the conversion method take new diags, and convert them to old diags.

Reply:
Theoretically, this should be the `statuspage` that we end up using in the long run anyway. And the old one should go away. I think the duplication makes sense when all of the products convert and we can remove the old code. And I don't know if we gain a lot by having a method to convert diagnostics, if this is one of the only methods we have to do it for.

Reply:
I think it will take a long time to convert all resources, so I would prioritize minimal duplication where possible. Converting diagnostics should be trivial, there may even be a method for it already in the mux package somewhere.